test/tools/yulInterpreter/Interpreter.h

```diff
@@ -35,7 +35,15 @@ namespace yul
 namespace test
 {
 
-class InterpreterTerminated: dev::Exception
+class ExplicitlyTerminated: dev::Exception
```

Comment:
If you keep `InterpreterTerminated` and make it a common base class of the other three, then it is enough to catch `InterpreterTerminated` if you do not care about how it was terminated.

Reply:
This is a bit tricky so let me write my notes down:

The fuzz harness does not care if the interpreter
  - was explicitly terminated via stop etc.
  - was terminated because trace limit was exceeded

The assumption behind not caring is that traces should be the same in these cases. The first case (explicit termination) is relatively straightforward. However, I tried to convince myself that the second case also is, but I'm not sure. Essentially, the question is the following: Do traces in the unoptimized and optimized code match exactly at all times? What if there is a side-effect in some dead code that was optimized out?

The test harness does care about
  - termination caused because of exceeding step limit

This is because of the example above.

Reply:
Please see https://github.com/ethereum/solidity/pull/6398/files#r271915747 for an example of what I meant.

Reply:
I think the link is not right, or at least it doesn't point to a code example